### PR TITLE
Wait for reads to finish before reading the result

### PR DESCRIPTION
**Pull request by euantorano**
As requested by @dom96, this fixes an issue seen here: https://github.com/nim-lang/redis/pull/4#issuecomment-312713921

### CODE REVIEW

**Comment by dom96** [comment]
I'm happy with this change, but a test would be nice to ensure this doesn't break. Would you be okay with creating a test for this?

**Comment by euantorano** [comment]
Sure, I’ll find some time this weekend.

**Comment by euantorano** [comment]
@dom96 Added a basic test, hopefully that's the kind of thing you were looking for?

File: tests/async/tasyncRecvLine.nim [hunk 1]
```diff
@@ -0,0 +1,54 @@
+discard """
+  file: "tasyncsend4754.nim"
```
Comment (dom96) [inline tests/async/tasyncRecvLine.nim]: You forgot to change this :)

**Comment by euantorano** [comment]
Oops, I'll fix that. I haven't verified that it fails without the patch either, will have to check that.

**Comment by euantorano** [comment]
Fixed the header, and made sure the test fails when the patch isn't in place:

```
Hint: C:\Users\euant\Desktop\Nim\tests\async\tasyncRecvLine.exe  [Exec]
Traceback (most recent call last)
tasyncRecvLine.nim(53)   tasyncRecvLine
asyncdispatch.nim(271)   waitFor
asyncdispatch.nim(326)   poll
asyncdispatch.nim(180)   processPendingCallbacks
asyncdispatch.nim(1271)  :anonymous
asyncfutures.nim(118)    complete
asyncmacro.nim(34)       cb
asyncmacro.nim(334)      sendIter
asyncfutures.nim(118)    complete
asyncmacro.nim(34)       cb
tasyncRecvLine.nim(31)   testUnbufferedIter
asyncmacro.nim(387)      recvLine
asyncmacro.nim(34)       cb
asyncnet.nim(587)        recvLineIter
asyncmacro.nim(387)      recvLineInto
asyncmacro.nim(34)       cb
asyncnet.nim(537)        recvLineIntoIter
asyncfutures.nim(225)    read
[[reraised from:
tasyncRecvLine.nim(53)   tasyncRecvLine
asyncdispatch.nim(271)   waitFor
asyncdispatch.nim(326)   poll
asyncdispatch.nim(180)   processPendingCallbacks
asyncmacro.nim(34)       cb
asyncmacro.nim           recvLineIter
asyncfutures.nim(220)    read
]]
[[reraised from:
tasyncRecvLine.nim(53)   tasyncRecvLine
asyncdispatch.nim(271)   waitFor
asyncdispatch.nim(326)   poll
asyncdispatch.nim(180)   processPendingCallbacks
asyncmacro.nim(50)       cb
asyncfutures.nim(156)    fail
asyncmacro.nim(34)       cb
asyncmacro.nim(123)      testUnbufferedIter
asyncfutures.nim(220)    read
]]
[[reraised from:
tasyncRecvLine.nim(53)   tasyncRecvLine
asyncdispatch.nim(1345)  waitFor
asyncfutures.nim(220)    read
]]
Error: unhandled exception: Future still in progress.
  recvLineInto's lead up to read of failed Future:
    Traceback (most recent call last)
    tasyncRecvLine.nim(53)   tasyncRecvLine
    asyncdispatch.nim(271)   waitFor
    asyncdispatch.nim(326)   poll
    asyncdispatch.nim(180)   processPendingCallbacks
    asyncdispatch.nim(1271)  :anonymous
    asyncfutures.nim(118)    complete
    asyncmacro.nim(34)       cb
    asyncmacro.nim(334)      sendIter
    asyncfutures.nim(118)    complete
    asyncmacro.nim(34)       cb
    tasyncRecvLine.nim(31)   testUnbufferedIter
    asyncmacro.nim(387)      recvLine
    asyncmacro.nim(34)       cb
    asyncnet.nim(587)        recvLineIter
    asyncmacro.nim(387)      recvLineInto
    asyncmacro.nim(34)       cb
    asyncnet.nim(537)        recvLineIntoIter
    asyncfutures.nim(225)    read
  recvLine's lead up to read of failed Future:
    Traceback (most recent call last)
    tasyncRecvLine.nim(53)   tasyncRecvLine
    asyncdispatch.nim(271)   waitFor
    asyncdispatch.nim(326)   poll
    asyncdispatch.nim(180)   processPendingCallbacks
    asyncdispatch.nim(1271)  :anonymous
    asyncfutures.nim(118)    complete
    asyncmacro.nim(34)       cb
    asyncmacro.nim(334)      sendIter
    asyncfutures.nim(118)    complete
    asyncmacro.nim(34)       cb
    tasyncRecvLine.nim(31)   testUnbufferedIter
    asyncmacro.nim(387)      recvLine
    asyncmacro.nim(34)       cb
    asyncnet.nim(587)        recvLineIter
    asyncmacro.nim(387)      recvLineInto
    asyncmacro.nim(34)       cb
    asyncnet.nim(537)        recvLineIntoIter
    asyncfutures.nim(225)    read
    [[reraised from:
    tasyncRecvLine.nim(53)   tasyncRecvLine
    asyncdispatch.nim(271)   waitFor
    asyncdispatch.nim(326)   poll
    asyncdispatch.nim(180)   processPendingCallbacks
    asyncmacro.nim(34)       cb
    asyncmacro.nim           recvLineIter
    asyncfutures.nim(220)    read
    ]]
  testUnbuffered's lead up to read of failed Future:
    Traceback (most recent call last)
    tasyncRecvLine.nim(53)   tasyncRecvLine
    asyncdispatch.nim(271)   waitFor
    asyncdispatch.nim(326)   poll
    asyncdispatch.nim(180)   processPendingCallbacks
    asyncdispatch.nim(1271)  :anonymous
    asyncfutures.nim(118)    complete
    asyncmacro.nim(34)       cb
    asyncmacro.nim(334)      sendIter
    asyncfutures.nim(118)    complete
    asyncmacro.nim(34)       cb
    tasyncRecvLine.nim(31)   testUnbufferedIter
    asyncmacro.nim(387)      recvLine
    asyncmacro.nim(34)       cb
    asyncnet.nim(587)        recvLineIter
    asyncmacro.nim(387)      recvLineInto
    asyncmacro.nim(34)       cb
    asyncnet.nim(537)        recvLineIntoIter
    asyncfutures.nim(225)    read
    [[reraised from:
    tasyncRecvLine.nim(53)   tasyncRecvLine
    asyncdispatch.nim(271)   waitFor
    asyncdispatch.nim(326)   poll
    asyncdispatch.nim(180)   processPendingCallbacks
    asyncmacro.nim(34)       cb
    asyncmacro.nim           recvLineIter
    asyncfutures.nim(220)    read
    ]]
    [[reraised from:
    tasyncRecvLine.nim(53)   tasyncRecvLine
    asyncdispatch.nim(271)   waitFor
    asyncdispatch.nim(326)   poll
    asyncdispatch.nim(180)   processPendingCallbacks
    asyncmacro.nim(50)       cb
    asyncfutures.nim(156)    fail
    asyncmacro.nim(34)       cb
    asyncmacro.nim(123)      testUnbufferedIter
    asyncfutures.nim(220)    read
    ]] [ValueError]
Error: execution of an external program failed: 'C:\Users\euant\Desktop\Nim\tests\async\tasyncRecvLine.exe '
```

**Comment by dom96** [comment]
Beautiful. Thanks @euantorano :D

**Pull request merged**